diff --git a/node_modules/@react-native-community/checkbox/typings/index.d.ts b/node_modules/@react-native-community/checkbox/typings/index.d.ts
index b48b503..e417fbf 100644
--- a/node_modules/@react-native-community/checkbox/typings/index.d.ts
+++ b/node_modules/@react-native-community/checkbox/typings/index.d.ts
@@ -1,12 +1,14 @@
 import {Component} from 'react';
-import {Constructor, NativeMethodsMixin} from 'react-native';
+import {NativeMethods} from 'react-native';
 import {Props as AndroidProps} from '../dist/CheckBox.android';
 import {Props as IOSProps} from '../dist/CheckBox.ios';
 import {Props as WindowsProps} from '../dist/CheckBox.windows';
 
+type Constructor<T> = new (...args: any[]) => T;
+
 type CheckBoxProps = AndroidProps & IOSProps & WindowsProps;
 
 declare class CheckBoxComponent extends Component<CheckBoxProps> {}
-declare const CheckBoxBase: Constructor<NativeMethodsMixin> &
+declare const CheckBoxBase: Constructor<NativeMethods> &
   typeof CheckBoxComponent;
 export default class CheckBox extends CheckBoxBase {}
