diff --git a/node_modules/react-native-linear-gradient/index.android.js b/node_modules/react-native-linear-gradient/index.android.js
index e35bee0..4d6304b 100644
--- a/node_modules/react-native-linear-gradient/index.android.js
+++ b/node_modules/react-native-linear-gradient/index.android.js
@@ -80,21 +80,23 @@ export default class LinearGradient extends Component<Props> {
       validRadius(flatStyle.borderBottomLeftRadius)
     ];
 
+
     return (
-      <View ref={(component) => { this.gradientRef = component; }} {...otherProps} style={style}>
-        <NativeLinearGradient
-          style={{position: 'absolute', top: 0, left: 0, bottom: 0, right: 0}}
-          colors={colors.map(processColor)}
-          startPoint={convertPoint('start', start)}
-          endPoint={convertPoint('end', end)}
-          locations={locations ? locations.slice(0, colors.length) : null}
-          useAngle={useAngle}
-          angleCenter={convertPoint('angleCenter', angleCenter)}
-          angle={angle}
-          borderRadii={borderRadiiPerCorner}
-        />
-        { children }
-      </View>
+      <NativeLinearGradient
+        ref={(component) => { this.gradientRef = component; }}
+        {...otherProps}
+        startPoint={convertPoint('start', start)}
+        endPoint={convertPoint('end', end)}
+        colors={colors.map(processColor)}
+        locations={locations ? locations.slice(0, colors.length) : null}
+        useAngle={useAngle}
+        angleCenter={convertPoint('angleCenter', angleCenter)}
+        angle={angle}
+        style={[{position: 'absolute', top: 0, left: 0, bottom: 0, right: 0}, style]}
+        borderRadii={borderRadiiPerCorner}
+      />
     );
+
+    
   }
 }
